module04/03-Panda-Social-Network/solution.py

```diff
@@ -0,0 +1,151 @@
+import re
+
+
+class Panda:
+    """Panda which is user of :class:`PandaSocialNetwork` and has basic attributes as name,
+    email, gender and list of friends"""
+
+    def __init__(self, name, email, gender):
+        self.validate_input(name, email, gender)
+        self.panda_name = name
+        self.panda_email = email
+        self.panda_gender = gender
+        self.friend_list = []
+
+    def __str__(self):
+        return "%s panda named %s with email: %s" % (self.panda_gender.capitalize(), self.panda_name, self.panda_email)
+
+    def __eq__(self, other):
+        if isinstance(other, Panda):
+            return self.__key() == other.__key()
+        raise NotImplemented
```

Comment:
tests are passing, because python does some magic under the hood, but I still think this should be `return False`, not `raise NotImplemented`